### ISSUE TITLE
GetManagedObjects fails after DestroyPool is called since blockdevs remain

### ISSUE THREAD

**Issue by mulkieran** [stratis-storage/stratisd#644]
This is occurring to the sim engine, which makes it likely that the crash is in the D-Bus layer.

It is almost certain that the crash is being caused by the failure to remove the blockdevs when the pool is removed.

See PR #628.

This bug is showing up, intermittently but reliably, in stratisd-dbus-client tests. Why intermittently? Because it is possible to create a pool with no devices on SimEngine. So, if the pool has no blockdevs, there is no trouble when they are not removed.

**Comment by mulkieran** [comment]
Much to my surprise though, the crash occurs in main(), when polling the file descriptors.

```
Program received signal SIGTERM, Terminated.
0x00007f90096e4000 in __poll_nocancel () from /lib64/libc.so.6
(gdb) bt
#0  0x00007f90096e4000 in __poll_nocancel () from /lib64/libc.so.6
#1  0x0000555c8902fe99 in stratisd::run () at src/bin/stratisd.rs:97
#2  0x0000555c890308ef in stratisd::main () at src/bin/stratisd.rs:118
#3  0x0000555c8964589d in __rust_maybe_catch_panic ()
#4  0x0000555c8963f0ca in std::rt::lang_start::he62d7f561d133c45 ()
#5  0x0000555c89030a73 in main ()
```

I had imagined it was a failed ```unwrap()``` in the D-Bus layer.

**Comment by mulkieran** [comment]
Better info. After destroy:

```
[root@localhost stratisd]# busctl call org.storage.stratis1 /org/storage/stratis1 org.freedesktop.DBus.ObjectManager GetManagedObjects
no path for parent object path /org/storage/stratis1/1
```

which is the object path of the pool. It's not clear where that message is coming from, though. But it is the correct thing.

**Comment by mulkieran** [comment]
We believe that this was fixed by #660.
